src/main/java/com/example/demo/service/shoppingcart/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,65 @@
+package com.example.demo.service.shoppingcart;
+
+import com.example.demo.dto.shoppingcart.CartItemRequestDto;
+import com.example.demo.dto.shoppingcart.ShoppingCartResponseDto;
+import com.example.demo.exception.EntityNotFoundException;
+import com.example.demo.mapper.CartItemMapper;
+import com.example.demo.mapper.ShoppingCartMapper;
+import com.example.demo.model.CartItem;
+import com.example.demo.model.ShoppingCart;
+import com.example.demo.repository.book.BookRepository;
+import com.example.demo.repository.shoppingcart.CartItemRepository;
+import com.example.demo.repository.shoppingcart.ShoppingCartRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final BookRepository bookRepository;
+    private final CartItemRepository cartItemRepository;
+    private final CartItemMapper cartItemMapper;
+    private final ShoppingCartMapper shoppingCartMapper;
+
+    @Override
+    public ShoppingCartResponseDto getCartByUserId(Long userId) {
+        ShoppingCart cart = shoppingCartRepository.findByUserId(userId)
+                .orElseThrow(() -> new EntityNotFoundException(
+                        "Shopping cart not found for user id: " + userId));
+        return shoppingCartMapper.toDto(cart);
+    }
+
+    @Override
+    public ShoppingCartResponseDto addBookToCart(
+            Long userId, CartItemRequestDto cartItemRequestDto) {
+        ShoppingCart cart = shoppingCartRepository.findByUserId(userId)
+                .orElseThrow(() -> new EntityNotFoundException(
+                        "Shopping cart not found for user id: " + userId));
+        bookRepository.findById(cartItemRequestDto.bookId())
+                .orElseThrow(() -> new EntityNotFoundException(
+                        "Book not found with id: " + cartItemRequestDto.bookId()));
+
+        CartItem cartItem = cartItemMapper.toEntity(cartItemRequestDto);
```

Comment:
check if cart already includes book, if yes - increase quantity, no - create new cart item

```suggestion
        cart.getCartItems()..... 
```